src/views/CashlinkManage.vue

```diff
@@ -93,20 +93,33 @@ export default class CashlinkManage extends Vue {
     private readonly nativeShareAvailable: boolean = (!!window.navigator && !!window.navigator.share);
 
     private async mounted() {
-        this.retrievedCashlink = this.cashlink || null;
-        if (!this.keyguardResult && this.request.cashlinkAddress) {
-            const cashlink = await CashlinkStore.Instance.get(this.request.cashlinkAddress.toUserFriendlyAddress());
-            if (cashlink) {
-                this.retrievedCashlink = cashlink;
-                this.isTxSent = true;
-                this.isManagementRequest = true;
-            } else {
-                this.$rpc.reject(new Error(`${this.request.cashlinkAddress} is not a valid Cashlink.`));
+        let storedCashlink;
+        if (this.request.cashlinkAddress) {
+            storedCashlink = await CashlinkStore.Instance.get(this.request.cashlinkAddress.toUserFriendlyAddress());
+            if (!storedCashlink) {
+                this.$rpc.reject(new Error(`Could not find Cashlink for address ${this.request.cashlinkAddress}.`));
                 return;
             }
+        } else if (this.cashlink) {
+            storedCashlink = await CashlinkStore.Instance.get(this.cashlink.address.toUserFriendlyAddress());
+        } else {
+            // this can happen if the user reloads the page after coming from SignTransactionLedger
+            history.back();
+            return;
+        }
+
+        if (storedCashlink) {
+            // cashlink already sent and stored
+            this.retrievedCashlink = storedCashlink;
+            this.isTxSent = true;
+            this.isManagementRequest = !this.cashlink; // freshly created cashlink or management request?
+        } else {
+            this.retrievedCashlink = this.cashlink!;
         }
```

Comment:
That's not correct. See comment above.

Reply:
After the keyguard returns to this code, the cashlink will first enter the `CashlinkStore` a couple of lines down in Line 144. Thus @danimoh is correct in saying that `storedCashlink` will be `null` here in that case.

However, if the browser crashes during sending (which is why the storing part was moved up) and the page is reloaded it will no longer be `null`, but a transaction was not send.

Reply:
The main target to move the storing up was to avoid funds getting lost by a transaction that was sent to the network without storing the Cashlink. This is still met by the current implementation.

Additionally, harm to the user experience / user confusion is limited as the Cashlink will not pop up in the safe as there is no matching transaction. When he then sees that his balance did not change and there is no Cashlink charging transaction, he can reason, that no Cashlink was created and attempt to do so again.

In the previous implementation, the transaction signature was available after reloads because the keyguard result is written to session storage in the rpc client. This might or might not be available after a browser crash depending on whether the user configured his browser to resume sessions and whether the browser already wrote that information to disk. Thus, also the previous implementation was not completely fail safe.

However if the transaction failed to be sent without the browser crashing, the previous implementation was able to resend the transaction on reload.
I could archive a similar effect by storing the signed transaction in the history state.

Alternatively, we could think about storing transactions that have not been sent to the network yet in localstorage and then trying to send them out to the network whenever the hub is opened with a request that stays in the hub (i.e. does not simply redirect to the keyguard). That would be a more sophisticated solution as it would also work after the window is closed, also for transmission of checkout transactions.
Transactions that reach the end of their validity duration would get pruned.
On the other hand, this might feel confusing to the user if a transaction that failed in his opinion is later sent to the network.
Thus, maybe keeping the tx only until the current tab is closed, is actually preferrable?

Reply:
I would agree that transaction should be send in the flow they are created in, not at some unspecified and unreliable point in time in the future.

I wanted to point out that there is a possibility, that this condition fails, as there is a disconnect in when a Cashlink enters the store and when it becomes functional. I am aware that no funds can be lost.

I agree that the harm done to the user is small and more an inconvenience than harm. 
However alerting the user to this inconvenience and not waiting for him to find out on its own might be preferable here. Could potentially be done in a separate PR, as it is not Ledger specific since, as you pointed out, the previous implementation was not fail-safe as well.

Reply:
I agree. Should be a separate PR to get this one merged faster.

Reply:
So this code fails in case the transaction sending fails/hangs/crashes the browser and the user reloads? Then it will look like it sent? Does it also look like it sent when I relead the page just as the consensus gets established before the tx sends?

Reply:
I want to say yes.

Reply:
The consensus is already established in `SignTransactionLedger`.
And we already said we will handle this in a separate PR.
I will resolve this conversation now.

Reply:
I'm not talking about Ledger-signed cashlinks, I'm talking about Keyguard-signed cashlinks. It sounds to me like you **introduced** a bug, where when I come back from the Keyguard after entering my password and I get the loading screen in the Hub here that *then establishes consensus here* and tries to send the funding transaction, and for any reason stops working and I reload the page, then it immediately shows me that the cashlink was funded successfully, although no transaction was ever sent.

Is my understanding correct? If so, I disapprove this PR because that is a serious new bug for me.

Reply:
Actually not sure how to best proceed here.
I kind of agree with @sisou that the behaviour especially for a reload should be working. However there is also a huge discussion about what to include and not include going on.

I personally would probably opt to restore the reload behaviour as it was working before, while moving the rest of the problem to a different PR. I assume that i.e. a crashed browser is sort of a good indicator, that something went wrong as well and handling that case could be postponed until later. Especially since no funds would be lost.

Reply:
I think the case where a cashlink funding transaction fails to send because consensus-establishing fails (after coming back from Keyguard) is way more common than creating a Cashlink with Ledger. So I would optimize for doing the expected thing when reloading the consensus-loading-screen.

Reply:
Resending of unrelayed transactions is addressed by https://github.com/nimiq/hub/pull/368